iroh-api/src/error.rs

```diff
@@ -0,0 +1,27 @@
+use std::io;
+use thiserror::Error as ThisError;
+// use std::error::Error;
```

Comment:
have you gotten the error talk from dig yet? don't want to get in trouble for approving `thiserror` usage on top of `anyhow` usage 🤣